File: sdks/python/bolt_prediction_sdk/models/energy_carbon_params_schema.py
"""
EnergyCarbonParamsSchema 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class EnergyCarbonParamsSchema(SDKBaseModel):
    """能耗与碳排增量模型参数"""

    energy_per_leakage_unit: Optional[float] = Field(description="单位泄漏能耗 (kWh/m³)", default=8.5)
    carbon_factor_electricity: Optional[float] = Field(description="电力排放因子 (kgCO₂e/kWh)", default=0.5839)
    carbon_factor_natural_gas: Optional[float] = Field(description="天然气排放因子 (kgCO₂e/kWh)", default=2.1622)
    carbon_factor_steam: Optional[float] = Field(description="蒸汽排放因子 (kgCO₂e/kWh)", default=0.11)
    compressor_efficiency: Optional[float] = Field(description="压缩机效率 0-1", default=0.75)
    recovery_rate: Optional[float] = Field(description="泄漏回收率 0-1", default=0.0)
    base_monthly_energy_kwh: Optional[float] = Field(description="基准月度能耗 (kWh)", default=10000.0)
    base_monthly_carbon_kg: Optional[float] = Field(description="基准月度碳排 (kgCO₂e)", default=5839.0)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/EnergyCarbonParamsSchema.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 能耗与碳排增量模型参数 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class EnergyCarbonParamsSchema {

    @JsonProperty("energy_per_leakage_unit")
    private Double energyPerLeakageUnit;

    @JsonProperty("carbon_factor_electricity")
    private Double carbonFactorElectricity;

    @JsonProperty("carbon_factor_natural_gas")
    private Double carbonFactorNaturalGas;

    @JsonProperty("carbon_factor_steam")
    private Double carbonFactorSteam;

    @JsonProperty("compressor_efficiency")
    private Double compressorEfficiency;

    @JsonProperty("recovery_rate")
    private Double recoveryRate;

    @JsonProperty("base_monthly_energy_kwh")
    private Double baseMonthlyEnergyKwh;

    @JsonProperty("base_monthly_carbon_kg")
    private Double baseMonthlyCarbonKg;

    public Double getEnergyPerLeakageUnit() {
        return energyPerLeakageUnit;
    }

    public void setEnergyPerLeakageUnit(Double energyPerLeakageUnit) {
        this.energyPerLeakageUnit = energyPerLeakageUnit;
    }

    public Double getCarbonFactorElectricity() {
        return carbonFactorElectricity;
    }

    public void setCarbonFactorElectricity(Double carbonFactorElectricity) {
        this.carbonFactorElectricity = carbonFactorElectricity;
    }

    public Double getCarbonFactorNaturalGas() {
        return carbonFactorNaturalGas;
    }

    public void setCarbonFactorNaturalGas(Double carbonFactorNaturalGas) {
        this.carbonFactorNaturalGas = carbonFactorNaturalGas;
    }

    public Double getCarbonFactorSteam() {
        return carbonFactorSteam;
    }

    public void setCarbonFactorSteam(Double carbonFactorSteam) {
        this.carbonFactorSteam = carbonFactorSteam;
    }

    public Double getCompressorEfficiency() {
        return compressorEfficiency;
    }

    public void setCompressorEfficiency(Double compressorEfficiency) {
        this.compressorEfficiency = compressorEfficiency;
    }

    public Double getRecoveryRate() {
        return recoveryRate;
    }

    public void setRecoveryRate(Double recoveryRate) {
        this.recoveryRate = recoveryRate;
    }

    public Double getBaseMonthlyEnergyKwh() {
        return baseMonthlyEnergyKwh;
    }

    public void setBaseMonthlyEnergyKwh(Double baseMonthlyEnergyKwh) {
        this.baseMonthlyEnergyKwh = baseMonthlyEnergyKwh;
    }

    public Double getBaseMonthlyCarbonKg() {
        return baseMonthlyCarbonKg;
    }

    public void setBaseMonthlyCarbonKg(Double baseMonthlyCarbonKg) {
        this.baseMonthlyCarbonKg = baseMonthlyCarbonKg;
    }

}